### ISSUE TITLE
Error in compilation -  Thoth.Json requires a value ValLinkagePartialKey(generateEncoder)

### ISSUE THREAD

**Issue by tforkmann** [thoth-org/Thoth.Json#46]
Hi Maxime,

I get following error message with Thoth.Json 3.4.1:

FSC : error FS0193: Das Modul bzw. der Namespace "Encode" aus der Kompilierungseinheit "Thoth.Json" enthielt nicht den Wert "ValLinkagePartialKey(generateEncoder)". [C:\Users\tforkmann\Documents\1_Tests\PIvy\src\Client

Downgrading to Thoth.Json 3.3.0 solved the issue.

Could you maybe help me solving this error?

Thanks in advance!

**Comment by MangelMaxime** [comment]
Duplicate of #45 

Can you please try to use `Fable.Elmish.Debugger 3.1.0-beta-1`?

**Comment by tforkmann** [comment]
Sorry for the late response!

Works! Thanks

**Comment by tforkmann** [comment]
Hi there,

I'm getting the same issue again. Using FSharp.Core 5.0, .NET5.0 and running on Mac.

```
TimsMacBookPro:src tforkmann$ dotnet watch run
watch : Started
FSC : error FS0193: The module/namespace 'Encode' from compilation unit 'Thoth.Json' did not contain the val 'ValLinkagePartialKey(toString)'
```

**Comment by MangelMaxime** [comment]
That's strange I didn't release a new version since a long time.

Last time it was because, I release some version which didn't have a breaking change in the exposed API but still .Net considered it broken...

Does it mean that because FSharp.Core has received an update it breaks existing library?

**Comment by forki** [comment]
I think it needs to be reported to F# Compiler. @cartermp

**Comment by cartermp** [comment]
Yes, this is an internal F# compiler error, but this sounds like something related to the F# metadata format, which is not forwards-compatible. But we'd have to investigate further. Could you file this under `dotnet/fsharp` with an up-to-date repro?

**Comment by MangelMaxime** [comment]
@tforkmann What do I need to do right now to unblock the situation?

Did you try to build Thoth.Json with latest FSharp.Core version and see if it fixed it?

**Comment by tforkmann** [comment]
@MangelMaxime I forced the FSharp.Core version 5.0.0 in my dependencies like this:
`nuget FSharp.Core == 5.0.0`

Then the compiler issue is disappearing.
So I'm unblocked for now.

@cartermp Will try to create a repro and then create an issue under `dotnet/fsharp`
